app/controllers/comments_controller.rb

```diff
@@ -1,2 +1,19 @@
 class CommentsController < ApplicationController
+  def create
```

Comment:
- [ ] Good job so far 👏  but According to the project requirement user needs to be able to create a comment on behalf of the current_user, for that reason it should have a method **new**  to get a form, and a form  **views/comments/new.html.erb** file with input for creating comments. Please fix it in order to make your app works as expected 👍

Reply:
Hi @codecaiine I get your point according to get a new view in `app/views/comments/new.html.erb` and the method `new` in `CommentsController` it's not listed in the requirements, it just says:  `Create a comment on behalf of the current_user`. So it doesn't mention that it's strictly necessary, but it's necessary to create a comment on behalf of the current_user.
It's been fulfilled in the method `create` where I am assigning `current_user` to `@comment.user`:
```
def create
    @post = Post.find(params[:post_id])
    @comment = @post.comments.new(comment_params)
    @comment.user = current_user
    if @comment.save
      @comment.update_counter
      redirect_to user_post_path(current_user.id, @post.id), notice: 'You created a comment!'
    else
      render :new, status: :unprocessable_entity
    end
  end
  ```

Reply:
Hi, @anderCM  Thank you to reach out to me. In the project requirement, you are supposed to create a form for both POST AND COMMENT and In order to get this form you should have a method **def new** inside the comment controller like the one you did in your Post controller.

# controllers/comments_controller.rb

class CommentsController < ApplicationController
  ...

  def new
    @comment = Comment.new
  end

def create
   /// Your code 
  end

 
	...
end

Reply:
@codecaiine  Thank you for clarifying your point. I will work on that